src/components/screens/HistoryScreen/mappers/payment.tsx

```diff
@@ -56,7 +104,15 @@ export const mapPaymentHistoryItem = ({
     from,
   } = operation;
 
-  const isRecipient = to === publicKey && from !== publicKey;
+  // Extract actual destination from XDR (may be M address)
+  // Horizon API returns base G address, but XDR contains the actual address used
+  const actualDestination = extractDestinationFromXDR(xdr, network, to);
+
+  // If destination is an M address, clear memo (memo is encoded in the address)
+  const isDestinationMuxed = isMuxedAccount(actualDestination);
+  const finalMemo = isDestinationMuxed ? "" : memo;
```

Comment:
we should hide only for Soroban, as mentioned on the o[ther comment](https://github.com/stellar/freighter-mobile/pull/597/files#r2577623212)